LEGO1/lego/legoomni/src/race/carrace.cpp

```diff
@@ -1,60 +1,418 @@
+		else if (m_unk0x14c == objectId) {
+			NotificationManager()->Send(this, MxNotificationParam());
+		}
+	}
+
+	return result;
 }
 
-// STUB: LEGO1 0x100170e0
-MxLong CarRace::HandlePathStruct(LegoPathStructNotificationParam&)
+// FUNCTION: LEGO1 0x100170e0
+// FUNCTION: BETA10 0x100c87ac
+MxLong CarRace::HandlePathStruct(LegoPathStructNotificationParam& p_param)
 {
-	// TODO
-	return 0;
+	MxLong result = 0;
+
+	if (p_param.GetTrigger() == 68) {
+		MxEntity* sender = (MxEntity*) p_param.GetSender();
+		MxS32 paramData = p_param.GetData();
+
+		switch (sender->GetEntityId()) {
+		case 10:
+			if (paramData <= m_unk0x104 || paramData >= m_unk0x104 + 5) {
+				break;
+			}
+
+			m_unk0x104 = paramData;
+			LegoChar buffer[20];
+			sprintf(buffer, "%g", 0.036 + 0.928 * (m_unk0xf8 * 20.0 + m_unk0x104) / (g_unk0x100f0c7c * 20.0));
+			VariableTable()->SetVariable("DISTANCE", buffer);
+
+			if (m_unk0x104 == 0x14) {
+				m_unk0x104 = 0;
+				m_unk0xf8++;
+
+				if (g_unk0x100f0c7c == m_unk0xf8) {
+					VariableTable()->SetVariable(g_raceState, "");
+
+					m_unk0x110[0]->Mute(TRUE);
+					m_unk0x110[1]->Mute(TRUE);
+					m_unk0x110[2]->Mute(TRUE);
+
+					m_unk0x110[0]->SetMaxLinearVel(-1.0);
+					m_unk0x110[1]->SetMaxLinearVel(-1.0);
+					m_unk0x110[2]->SetMaxLinearVel(-1.0);
+
+					RemoveActor(m_unk0x110[1]);
+					m_unk0x110[1]->ClearMaps();
+
+					RemoveActor(m_unk0x110[2]);
+					m_unk0x110[2]->ClearMaps();
+
+					MxS32 position;
+
+					if (m_unk0xfc < m_unk0xf8 && m_unk0x100 < m_unk0xf8) {
+						position = 3;
+						m_unk0x148 = g_unk0x100d5d40[rand() % 3];
+						m_unk0x14c = g_unk0x100d5d60[rand() % 3];
+					}
+					else if (m_unk0xfc < m_unk0xf8 || m_unk0x100 < m_unk0xf8) {
+						position = 2;
+						if (m_unk0xfc == g_unk0x100f0c7c) {
+							m_unk0x148 = g_unk0x100d5d30[rand() % 4];
+							m_unk0x14c = g_unk0x100d5d60[rand() % 3];
+						}
+						else {
+							m_unk0x148 = g_unk0x100d5d50[rand() % 3];
+							m_unk0x14c = g_unk0x100d5d40[rand() % 3];
+						}
+					}
+					else {
+						position = 1;
+						m_unk0x148 = g_unk0x100d5d30[rand() % 4];
+						m_unk0x14c = g_unk0x100d5d50[rand() % 3];
+					}
+
+					InputManager()->Reset88and336();
+					InputManager()->SetUnknown336(TRUE);
+					VariableTable()->SetVariable(g_strHIT_WALL_SOUND, "");
+					NavController()->SetDeadZone(NavController()->GetDefaultDeadZone());
+					NavController()->SetTrackDefault(1);
+					LegoRaceCar::FUN_10012de0();
+					m_raceState->m_unk0x28 = 2;
+
+					RaceState::Entry* raceState = m_raceState->GetState(GameState()->GetActorId());
+					raceState->m_unk0x02 = position;
+
+					if (raceState->m_score < (MxS16) position) {
+						raceState->m_score = position;
+					}
+
+					AnimationManager()->FUN_10060dc0(m_unk0x148, NULL, TRUE, FALSE, NULL, FALSE, TRUE, FALSE, TRUE);
+				}
+
+				result = 1;
+			}
+
+			break;
+		case 11:
+			if (paramData <= m_unk0x108 || paramData >= m_unk0x108 + 5) {
+				break;
+			}
+
+			FUN_10017820(0xb, paramData);
```

Comment:
To be consistent I'd use the decimal `11` instead of `0xb` since the function itself uses those values (same with the other instance of that call)